### PR TITLE
[wip] - Feature/rewrite tests/master

### PR DESCRIPTION
**Pull request by sensorario**

### CODE REVIEW

File: src/Mado/QueryBundle/Interfaces/EntityClass.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Mado\QueryBundle\Interfaces;
+
+interface EntityClass
+{
+    public function getFQNC();
```
Comment (AlessandroMinoccheri) [inline src/Mado/QueryBundle/Interfaces/EntityClass.php]: a better name?

File: src/Mado/QueryBundle/Queries/QueryBuilderFactory.php [hunk 2]
```diff
@@ -37,6 +37,11 @@ class QueryBuilderFactory extends AbstractQuery
 
     protected $select;
 
+    public function ensureFieldsDefinedPublic()
+    {
+        $this->ensureFieldsDefined();
```
Comment (AlessandroMinoccheri) [inline src/Mado/QueryBundle/Queries/QueryBuilderFactory.php]: Is better to render ensureFieldsDefined public I think

File: tests/Queries/QueryBuilderFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            'driver' => 'pdo_sqlite',
+            'path' => __DIR__ . '/data/db.sqlite',
+        ),
+        \Doctrine\ORM\Tools\Setup::createAnnotationMetadataConfiguration(
+            array(__DIR__."/src"),
+            true
+        ));
+    }
+
+    public function testCanFilterThanksToOperators()
```
Comment (AlessandroMinoccheri) [inline tests/Queries/QueryBuilderFactoryTest.php]: testCanFilterByOperators or testCanFilterWithOperators

File: tests/Queries/QueryBuilderFactoryTest.php [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            "WHERE g1_.name LIKE ? " .
+            "AND g1_.name LIKE ? " .
+            "AND g1_.name LIKE ? " .
+            "AND g1_.name LIKE ?",
+            $queryBuilderFactory->getQueryBuilder()->getQuery()->getSql()
+        );
+    }
+}
+
+/** @Entity() */
```
Comment (AlessandroMinoccheri) [inline tests/Queries/QueryBuilderFactoryTest.php]: Is better to move this into separate files?

File: tests/functional/FooTest.php [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                'filters' => [
+                    'name|contains|1' => 'foo',
+                    'name|contains|2' => 'bar',
+                ]
+            ]),
+            new Foo()
+        );
+
+        $this->assertEquals(
+            "SELECT f0_.id AS id_0, f0_.name AS name_1 FROM Foo f0_ WHERE f0_.name LIKE ? AND f0_.name LIKE ?",
```
Comment (AlessandroMinoccheri) [inline tests/functional/FooTest.php]: What do you think to create an assert to compare filter array with what you pass into the request?
Same in many parts of the code

File: tests/functional/FooTest.php [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+            new Foo()
+        );
+
+        $this->assertEquals(
+            "SELECT f FROM Foo f",
+            $dql->getDql()
+        );
+    }
+
+    public function testFooooo()
```
Comment (AlessandroMinoccheri) [inline tests/functional/FooTest.php]: change name function

File: src/Mado/QueryBundle/Interfaces/EntityClass.php (hunk 1 above)
Comment (sensorario) [reply src/Mado/QueryBundle/Interfaces/EntityClass.php]: Is an idiomatic name that stands for FullyQualifiedClassName, is this better thatn FQCN?

File: tests/Queries/QueryBuilderFactoryTest.php (hunk 3 above)
Comment (sensorario) [reply tests/Queries/QueryBuilderFactoryTest.php]: ByOperator is better

File: tests/functional/FooTest.php (hunk 6 above)
Comment (sensorario) [reply tests/functional/FooTest.php]: Yes 😨

File: tests/functional/FooTest.php (hunk 5 above)
Comment (sensorario) [reply tests/functional/FooTest.php]: There is no request here

File: src/Mado/QueryBundle/Interfaces/EntityClass.php (hunk 1 above)
Comment (AlessandroMinoccheri) [reply src/Mado/QueryBundle/Interfaces/EntityClass.php]: Maybe

File: tests/Queries/QueryBuilderFactoryTest.php (hunk 3 above)
Comment (AlessandroMinoccheri) [reply tests/Queries/QueryBuilderFactoryTest.php]: Ok ByOperator

File: tests/functional/FooTest.php (hunk 5 above)
Comment (AlessandroMinoccheri) [reply tests/functional/FooTest.php]: Yeah, but It's possible to ad an assert on filter array generated inside the QueryBundle?

File: tests/functional/FooTest.php (hunk 6 above)
Comment (AlessandroMinoccheri) [reply tests/functional/FooTest.php]: 👍

File: src/Mado/QueryBundle/Queries/QueryBuilderFactory.php (hunk 2 above)
Comment (sensorario) [reply src/Mado/QueryBundle/Queries/QueryBuilderFactory.php]: render?

Comment (sensorario) [reply src/Mado/QueryBundle/Queries/QueryBuilderFactory.php]: ah ok, sorry

File: tests/Queries/QueryBuilderFactoryTest.php (hunk 4 above)
Comment (sensorario) [reply tests/Queries/QueryBuilderFactoryTest.php]: What do you think about this folding structure:

tests/
└── resources
 |  └── ResourceName.php
└── unitary
 |  └── SomeUnitaryTest.php
└── integration
    └── SomeIntegrationTest.php

???

File: tests/functional/FooTest.php (hunk 5 above)
Comment (sensorario) [reply tests/functional/FooTest.php]: Could you explain better?

**Pull request closed without merging**